data/data/gcp/network/variables.tf

```diff
@@ -29,3 +29,15 @@ variable "network_cidr" {
 variable "worker_subnet_cidr" {
   type = string
 }
+
+variable "preexisting_cluster_network" {
```

Comment:
the prefix `preexisting` doesn't seem right.

```tf
variable "cluster_network" {
type = string
description = "The network for the cluster, if empty a new network will be created."
}
```

Reply:
Also needs to make sure if this is self_url or just name?

Reply:
> Also needs to make sure if this is self_url or just name?

I am confused about this. Do we want to allow passing both self_url and name in install-config? To me, name alone seems sufficient and easier for both us and users(how do you get self url of a resource you created?). If we need the self-url we can use the name to pull it with `data`

My take is that we just pass the name and then validate it. I _think_ this handles that scenario. If not or more needs to be done let's discuss so I can get a better handle.

Reply:
look at https://github.com/openshift/installer/pull/2406/files#diff-154c3e9ab591ac1381679f99d7bd3580R5 for example the output is self_url on case A, while this variable's value on case B.. so there should be right expectation in terms of what this needs to be.

I would want the install-config to have names, I agree.. but this comment was more to do with inconsistent use in the module and setting the expectation right.

Reply:
Got it. The input in tfvars is a name, and the input into the network module is a name, but the output of network module should be the same type (self-url) for both cases.